### PR TITLE
[refactor] alert reconciler

### PR DESCRIPTION
**Pull request by nicolastakashi**
This pull request primarily updates the Go version from 1.20 to 1.22 and makes several changes to the codebase to improve its efficiency and readability. The major changes include updating the Go version in multiple GitHub workflow files and the Dockerfile, updating the Controller Tools version in the Makefile, refactoring the `AlertSpec` and `Alert` structs in `alert_types.go`, and making changes to the `AlertReconciler` struct in `alert_controller.go`.

Updates to Go version:
* [`.github/workflows/build.yml`](diffhunk://#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721L25-R25): Updated Go version from 1.20.x to 1.22.x in the GitHub workflow.
* [`.github/workflows/docs.yml`](diffhunk://#diff-9cf2000c53760d837a449f874e53f792819108d3a4bf346336d0f7d082deae2cL24-R24): Updated Go version from 1.20.x to 1.22.x in the GitHub workflow.
* [`.github/workflows/e2e-tests.yaml`](diffhunk://#diff-deaabb10671a187e9d8bc2a25d16791aa368a5a6e3d970c3d42977712b6fd60fL33-R33): Updated Go version from 1.20.x to 1.22.x in the GitHub workflow.
* [`Dockerfile`](diffhunk://#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557L2-R2): Updated Go version from 1.20 to 1.22 in the Dockerfile.
* [`go.mod`](diffhunk://#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6L3-R3): Updated Go version from 1.20 to 1.22.

Updates to Makefile:
* [`Makefile`](diffhunk://#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52L143-R143): Updated the Controller Tools version from v0.9.2 to v0.15.0.

Refactoring of `AlertSpec` and `Alert` structs:
* [`apis/coralogix/v1alpha1/alert_types.go`](diffhunk://#diff-65ac8ce7bcca552a9cd61891b1088371d0f41bd94df1458dae2086ababbb5721L156-R175): Refactored the `AlertSpec` struct to simplify the `ExtractCreateAlertRequest` method and changed the receiver from `in *AlertSpec` to `a *Alert`. Added a `NewAlert` method to the `Alert` struct. [[1]](diffhunk://#diff-65ac8ce7bcca552a9cd61891b1088371d0f41bd94df1458dae2086ababbb5721L156-R175) [[2]](diffhunk://#diff-65ac8ce7bcca552a9cd61891b1088371d0f41bd94df1458dae2086ababbb5721R3110-R3117)

Changes to `AlertReconciler` struct:
* [`controllers/alphacontrollers/alert_controller.go`](diffhunk://#diff-4e859b1b1b51349871f27d1cad3387c630c5cf0d34f0c8e432dbafd8ad7377ebL27-R27): Made several changes to the `AlertReconciler` struct, including replacing the `jsonpb.Marshaler` with `logr.Logger`, refactoring the `Reconcile` method, and updating the `flattenAlert` function to return a non-pointer `AlertStatus`. [[1]](diffhunk://#diff-4e859b1b1b51349871f27d1cad3387c630c5cf0d34f0c8e432dbafd8ad7377ebL27-R27) [[2]](diffhunk://#diff-4e859b1b1b51349871f27d1cad3387c630c5cf0d34f0c8e432dbafd8ad7377ebL68-R232) [[3]](diffhunk://#diff-4e859b1b1b51349871f27d1cad3387c630c5cf0d34f0c8e432dbafd8ad7377ebL289-R272)

Updates to dependencies:
* [`go.mod`](diffhunk://#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6L18-R18): Updated the `k8s.io/utils` dependency version.

### CODE REVIEW

File: controllers/alphacontrollers/alert_controller.go [hunk 1]
```diff
@@ -76,177 +76,168 @@ type AlertReconciler struct {
...
+
+func (r *AlertReconciler) create(
+	ctx context.Context,
+	log logr.Logger,
+	alertClient clientset.AlertsClientInterface,
+	alert *coralogixv1alpha1.Alert) error {
+
+	if alert.Spec.Labels == nil {
+		alert.Spec.Labels = make(map[string]string)
+	}
```
[the remark this answers is not in the document]
Comment (nicolastakashi) [reply controllers/alphacontrollers/alert_controller.go:191]: Actually it is, after coming back from K8S API 😢

**Review by OrNovo — APPROVED** [review]

File: controllers/alphacontrollers/alert_controller.go [hunk 2]
```diff
@@ -65,188 +65,171 @@ type AlertReconciler struct {
...
 
-		createAlertReq, err := alertCRD.Spec.ExtractCreateAlertRequest(ctx)
-		if err != nil {
-			log.Error(err, "Bad request for creating alert", "Name", alertCRD.Name, "Namespace", alertCRD.Namespace)
-			return ctrl.Result{}, err
-		}
+	equal, _ := alert.Spec.DeepEqual(&status)
+	if equal {
+		return nil
+	}
```
Comment (matej-g) [inline controllers/alphacontrollers/alert_controller.go:141]: Micronit: Could be simplified

```suggestion
	if equal, _ := alert.Spec.DeepEqual(&status); equal {
		return nil
	}
```

File: controllers/alphacontrollers/alert_controller.go [hunk 3]
```diff
@@ -24,7 +24,7 @@ import (
 	"strconv"
 	"time"
 
-	"github.com/golang/protobuf/jsonpb"
+	"github.com/go-logr/logr"
```
Comment (matej-g) [inline controllers/alphacontrollers/alert_controller.go:27]: Is this a new dependency? Does it need to be added to `go.mod`?

**Review by matej-g — APPROVED** [review]
Nice job 👊 , two minor things!

**Review by matej-g — APPROVED** [review]

**Pull request merged**